app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheetController.kt

```diff
@@ -88,13 +95,44 @@ class ScheduleAppointmentSheetController @Inject constructor(
   }
 
   private fun schedulingSkips(events: Observable<UiEvent>): Observable<UiChange> {
-    return events.ofType<SchedulingSkipped>()
+    val patientUuidStream = events
+        .ofType<ScheduleAppointmentSheetCreated>()
+        .map { it.patientUuid }
+
+    val combinedStreams = Observables.combineLatest(
+        events.ofType<SchedulingSkipped>(),
+        patientUuidStream,
+        configProvider.toObservable()
+    )
+    val defaulterFeatureEnabledStream = combinedStreams
+        .filter { (_, _, config) -> config.isApiV3Enabled }
+        .switchMap { (_, patientUuid, config) ->
+          patientRepository
+              .isPatientDefaulter(patientUuid)
+              .flatMapSingle { isPatientDefaulter ->
+                when {
+                  isPatientDefaulter -> appointmentRepository
```

Comment:
What do you think of

```kotlin
val defaulterFeatureEnabledStream = combinedStreams
        .filter { (_, _, config) -> config.isApiV3Enabled }
        .switchMap { (_, patientUuid, config) ->
          val isPatientDefaulterStream = patientRepository
              .isPatientDefaulter(patientUuid)
              .replay()
              .refCount()

          val saveAppointmentAndCloseSheet = isPatientDefaulterStream
              .filter { isPatientADefaulter -> isPatientADefaulter }
              .flatMapSingle {
                appointmentRepository
                    .schedule(
                        patientUuid = patientUuid,
                        appointmentDate = LocalDate.now(utcClock).plus(config.appointmentDuePeriodForDefaulters),
                        isDefaulter = true)
              }
              .map { { ui: Ui -> ui.closeSheet() } }


          val closeSheetWithoutSavingAppointment = isPatientDefaulterStream
              .filter { isPatientADefaulter -> isPatientADefaulter.not() }
              .map { { ui: Ui -> ui.closeSheet() } }

          Observable.merge(saveAppointmentAndCloseSheet, closeSheetWithoutSavingAppointment)
        }
```

Reply:
It's more verbose than what was there earlier, but I also feel it's more readable. This might also fix the code complexity issue raised by Code Climate.

Reply:
There's a failing test which is also fixed by my implementation.

Reply:
> There's a failing test which is also fixed by my implementation.

Never mind, there's no failing test.

Reply:
This is more or less the same implementation. Instead of using `when`, this implementation uses `filter`.
Although, I will test these changes and see if it fixes the Code climate issue.